python/pyarrow/_compute.pyx

```diff
@@ -2036,15 +2036,16 @@ cdef class Expression(_Weakrefable):
 
     >>> import pyarrow.compute as pc
     >>> (pc.field("a") < pc.scalar(3)) | (pc.field("b") > 7)
-    <pyarrow.compute.Expression ((a < 3:int64) or (b > 7:int64))>
+    <pyarrow.compute.Expression ((a < 3) or (b > 7))>
+    >>> import pyarrow.dataset as ds
     >>> ds.field('a') != 3
```

Comment:
Since the line above is using `pc.field` instead of `ds.field`, can probably do that here as well (and avoid the extra import). This is from historical reasons, because this function was originally included in the datasets module, but then moved to the compute module (but it is still available in both places).